### PR TITLE
Refactor penalties

### PR DESCRIPTION
**Pull request by ivarnakken**
# Description

[Set default messages in select input](https://github.com/webkom/lego-webapp/commit/06c1c33a7d480d175be32c07db5b86391b7b3798)

You can witness these changes in the video below for a brief second hehe

---

[Refactor penalties code](https://github.com/webkom/lego-webapp/commit/1f41ba635bf791558e3123c0a167407e3e4a6cba) 

- Fix and improve validation. Actually, there was previously no validation
  at all since the so-called "message" field does not exist.
- Refactor to a functional component.
- Refactor to `react-final-form`.
- Adjust minor styling issues, such as spacing.
- Remove and clean up bad code
- Better ts types

---

[Update the list of penalties when deleting one](https://github.com/webkom/lego-webapp/commit/46221a3c71302c97307d9dc5f81be36775f69cb0) 

The store does not update accordingly, causing the list component to not
update either.

---

[Update the look of the penalty card](https://github.com/webkom/lego-webapp/commit/16310baa480a34a42ec038d9a333edbb57f5a193) 

The design might not be the best, but in my opinion it now looks more
"thorough" and improved. It had clearly not been given a lot of love in
the past.

Notice the small things such as the timeout on the "Prikken er
registrert!" feedback, or the now included info and link to the relevant
event.

Now that I look at the before and after, the changes seem very subtle ..

[Relevant backend change](https://github.com/webkom/lego/pull/3548)

# Result

<table>
	<tr>
		 <td>Before</td>
		 <td>After</td>
	<tr>
	 <td>
		<video src="https://github.com/webkom/lego-webapp/assets/69514187/d1aa7ac4-3fc1-4658-9ea7-4759b9722789" />
	</td>
	 <td>
		<video src="https://github.com/webkom/lego-webapp/assets/69514187/7d563ed0-0798-4cfa-a58e-d11a9fe9904d" />
	</td>
   </tr>
   <tr>
	 <td>
		image
	</td>
	 <td>
		image
	</td>
	<tr>
</table>

# Testing

- [x] I have thoroughly tested my changes.

Giving and removing penalties still work as expected.

---

Resolves ABA-645
Resolves ABA-722
Related to ABA-49

### CODE REVIEW

File: app/reducers/penalties.ts [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
 import createEntityReducer from 'app/utils/createEntityReducer';
 import { Penalty } from '../actions/ActionTypes';
 
+type State = any;
+const mutate = produce((newState: State, action: any): void => {
+  switch (action.type) {
+    case Penalty.DELETE.SUCCESS:
+      newState.items = without(newState.items, action.meta.penaltyId);
+      break;
+  }
```
[the remark this answers is not in the document]
Comment (ivarnakken) [reply app/reducers/penalties.ts:13]: Yes I see where you're coming from, but typically the mutate function looks at a bunch of different action types (look at e.g. [app/reducers/events.ts](https://github.com/webkom/lego-webapp/blob/master/app/reducers/events.ts)), and it those cases it's better to use switch statements. I figured it's best to stay consistent

[the remark this answers is not in the document]
Comment (itsisak) [reply app/reducers/penalties.ts:13]: I agree with staying consistent, not too messy anyways.

**Review by norbye — APPROVED** [review]

**Review by danielyanghansen — APPROVED** [review]

File: app/routes/users/components/Penalties.css [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+
+  &:hover {
+    color: var(--color-gray-6);
+  }
+}
+
+.info {
+  line-height: 1.3;
+
+  & :first-child {
```
Comment (haukkagu) [inline app/routes/users/components/Penalties.css:19]:
```suggestion
  &:first-child {
```

File: app/routes/users/components/Penalties.tsx [hunk 3]
```diff
@@ -51,16 +76,33 @@ function Penalties({ penalties, userId, canDeletePenalties }: Props) {
...
-        <span className="secondaryFontColor">Ingen prikker</span>
+        <>
+          <Flex alignItems="center" gap="1rem">
+            <Icon
+              name="thumbs-up-outline"
+              size={40}
+              className={styles.success}
+            />
+            <Flex column className={cx('secondaryFontColor', styles.info)}>
+              <span>Phuu ...</span>
```
Comment (haukkagu) [inline app/routes/users/components/Penalties.tsx:94]: Det staves faktisk [puh](https://naob.no/ordbok/puh), selv om det høres litt teitere ut

**Review by haukkagu — APPROVED** [review]
LGTM :)

File: app/routes/users/components/Penalties.css (hunk 2 above)
Comment (ivarnakken) [reply app/routes/users/components/Penalties.css:19]: Yea no. For reasons beyond science this does not work

**Pull request merged**